From: Takashi Iwai <tiwai@suse.de>
Subject: Workaround for kABI breakage by 4.4.67 iio hid-sensor changes
Patch-mainline: Never, openSUSE-42.2 only
References: stable-4.4.67

stable 4.4.67 update contains the changes for hid-sensor-hub that
breaks kABI.  Since the modified struct is embedded, we need to apply
trickery to shorten the new fields and put them into a hole.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/linux/hid-sensor-hub.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/include/linux/hid-sensor-hub.h
+++ b/include/linux/hid-sensor-hub.h
@@ -231,8 +231,10 @@ struct hid_sensor_common {
 	unsigned usage_id;
 	atomic_t data_ready;
 	atomic_t user_requested_state;
-	int poll_interval;
-	int raw_hystersis;
+#ifndef __GENKSYMS__
+	short poll_interval;
+	short raw_hystersis;
+#endif
 	struct iio_trigger *trigger;
 	struct hid_sensor_hub_attribute_info poll;
 	struct hid_sensor_hub_attribute_info report_state;
